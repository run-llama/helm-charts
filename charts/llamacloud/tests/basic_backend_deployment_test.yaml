suite: basic-backend-deployment

templates:
- ../templates/backend/deployment.yaml

tests:
- it: should be a Deployment
  set:
    backend.image.tag: latest
  asserts:
    - isKind:
        of: Deployment
    - equal:
        path: .spec.template.spec.containers[0].image
        value: llamaindex/llamacloud-backend:latest

- it: env vars should match snapshot for default values with subcharts enabled
  asserts:
    - matchSnapshot:
        path: .spec.template.spec.containers[0].env

- it: postgres env vars should use custom config if is subchart is not enabled and external config is provided
  set:
    postgresql.enabled: false
    global.config.postgresql.external.enabled: true
    global.config.postgresql.external.database: test-db
    global.config.postgresql.external.username: test-user
    global.config.postgresql.external.password: test-password
    global.config.postgresql.external.host: test-host
    global.config.postgresql.external.port: 5432
  asserts:
    - matchSnapshot:
        path: .spec.template.spec.containers[0].env