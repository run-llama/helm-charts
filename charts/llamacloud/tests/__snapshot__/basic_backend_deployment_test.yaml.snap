env vars should match snapshot for default values with subcharts enabled:
  1: |
    - name: LLAMACLOUD_LICENSE_KEY
      valueFrom:
        secretKeyRef:
          key: llamacloud-license-key
          name: RELEASE-NAME-llamacloud-license-key
    - name: DATABASE_HOST
      value: RELEASE-NAME-postgresql
    - name: DATABASE_PORT
      value: "5432"
    - name: DATABASE_NAME
      value: llamacloud
    - name: DATABASE_USER
      value: llamacloud
    - name: DATABASE_PASSWORD
      valueFrom:
        secretKeyRef:
          key: password
          name: RELEASE-NAME-postgresql
    - name: MONGODB_HOST
      value: RELEASE-NAME-mongodb
    - name: MONGODB_PORT
      value: "27017"
    - name: MONGODB_USER
      value: root
    - name: MONGODB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mongodb-root-password
          name: RELEASE-NAME-mongodb
    - name: JOB_QUEUE_ENDPOINT
      value: amqp://RELEASE-NAME-llamacloud-rabbitmq:5672
    - name: JOB_QUEUE_USERNAME
      value: user
    - name: JOB_QUEUE_PASSWORD
      valueFrom:
        secretKeyRef:
          key: rabbitmq-password
          name: RELEASE-NAME-llamacloud-rabbitmq
    - name: REDIS_HOST
      value: RELEASE-NAME-llamacloud-redis-master
    - name: REDIS_PORT
      value: "6379"
postgres env vars should use custom config if is subchart is not enabled and external config is provided:
  1: |
    - name: LLAMACLOUD_LICENSE_KEY
      valueFrom:
        secretKeyRef:
          key: llamacloud-license-key
          name: RELEASE-NAME-llamacloud-license-key
    - name: DATABASE_HOST
      value: test-host
    - name: DATABASE_PORT
      value: "5432"
    - name: DATABASE_NAME
      value: test-db
    - name: DATABASE_USER
      value: test-user
    - name: DATABASE_PASSWORD
      value: test-password
    - name: MONGODB_HOST
      value: RELEASE-NAME-mongodb
    - name: MONGODB_PORT
      value: "27017"
    - name: MONGODB_USER
      value: root
    - name: MONGODB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mongodb-root-password
          name: RELEASE-NAME-mongodb
    - name: JOB_QUEUE_ENDPOINT
      value: amqp://RELEASE-NAME-llamacloud-rabbitmq:5672
    - name: JOB_QUEUE_USERNAME
      value: user
    - name: JOB_QUEUE_PASSWORD
      valueFrom:
        secretKeyRef:
          key: rabbitmq-password
          name: RELEASE-NAME-llamacloud-rabbitmq
    - name: REDIS_HOST
      value: RELEASE-NAME-llamacloud-redis-master
    - name: REDIS_PORT
      value: "6379"
